### PR TITLE
Fixed issue #18995: Confidentiality issue about username

### PR DESCRIPTION
**Pull request by gabrieljenik**

### CODE REVIEW

File: application/models/services/SurveyUpdater/GeneralSettings.php [hunk 1]
```diff
@@ -90,6 +94,19 @@ public function update($surveyId, $input)
             throw new NotFoundException();
         }
 
+        // Before setting the owner, check if the user exists and can be seen
+        // by the current user (in case the request was forged)
+        // NOTE: Internally, the getUserList function will use objects (like the Yii App and Permission model) that
+        //       currently may differ from the ones injected in this service.
+        if (!empty($input['owner_id'])) {
+            $owner = $this->modelUser->findByPk($input['owner_id']);
+            if (!isset($owner) || !in_array($input['owner_id'], getUserList('onlyuidarray'))) {
```
Comment (Shnoulle) [inline application/models/services/SurveyUpdater/GeneralSettings.php:103]: On Service ? Not in controller or model ?

Comment (gabrieljenik) [reply application/models/services/SurveyUpdater/GeneralSettings.php:103]: Is a new pattern being implemented for placing all LS logic outside of framework, on own files.
This is the new "controller"-service on master. That's why we are adding it here.

Comment (Shnoulle) [reply application/models/services/SurveyUpdater/GeneralSettings.php:103]: OK :) 

Some other old function need to be move in Service then (DEV issue not needed)

File: application/models/services/SurveyUpdater/GeneralSettings.php [hunk 2]
```diff
@@ -90,6 +94,19 @@ public function update($surveyId, $input)
...
+                throw new PermissionDeniedException(
+                    'Permission denied'
```
Comment (Shnoulle) [inline application/models/services/SurveyUpdater/GeneralSettings.php:105]: Desired behavior : throw error, filter or send a model error ?

**Review by Shnoulle — COMMENTED** [review]
What is the desired behavior ?

The only way to set a invalid userId is to hack HTML or $_POST value ?

**Comment by Shnoulle** [comment]
Desired behavior : user try to hack server. 

Let's sysadmin check error log for such action. 

A 403 is the best !

File: application/models/services/SurveyUpdater/GeneralSettings.php (hunk 1 above)
Comment (gabrieljenik) [reply application/models/services/SurveyUpdater/GeneralSettings.php:103]: I guess that refactoring will slowly be happening with time. :)

**Comment by gabrieljenik** [comment]
Label "code review done" was added but approval is pending... :)

**Review by Shnoulle — APPROVED** [review]

**Pull request merged**